add Julius section (#702)
digital-asset/docs.daml.com: docs/3.1/docs/daml/reference/packages.rst

Because of their flexibility, data-dependencies are a tool that is recommended for performing Daml model upgrades. See the :ref:`upgrade documentation <upgrade-overview>` for more details.

Transitive dependency management
================================

> The Daml compiler identifies each DAR dependency in the project by their ``packageId`` and fully qualified name (Daml project package name and version number).

Author: ```suggestion
The Daml compiler identifies each DAR dependency in the project by its ``packageId`` and fully qualified name (Daml project package name and version number).
```

If you have a Daml project which contains multiple common transitive DAR dependencies, those common transitive dependencies must either:

> * Have the exact same contents if they have the same name and version specified in their Daml project's ``daml.yaml`` file, or

Reviewer: ```suggestion
* Have identical contents if they have the same name and version specified in their Daml project's ``daml.yaml`` file, or
```

> * Have a different version from each others' Daml project's ``daml.yaml`` file if there are differences in the contents.

Author: ```suggestion
* Have a different value for the ``version`` entry in their respective ``daml.yaml`` files.
```

> * Have a different version from each others' Daml project's ``daml.yaml`` file if there are differences in the contents.

Reviewer: ```suggestion
* Have different versions of the Daml project's ``daml.yaml`` file if there are differences in the contents.
```

Otherwise, the Daml project cannot be built into a deployable DAR due to package identification conflicts.

For example:

* Daml project X (top-level) has dependencies ``DarA`` and ``DarB``.
* ``DarA`` and ``DarB`` both contain DAR dependency ``DarC``

> When compiling Daml project X, you must ensure that the ``DarC`` dependency referenced by both ``DarA`` and ``DarB`` must either have identical Daml contents or have a different version number if their contents differ. The version number is defined in the daml.yaml file of the Daml project producing ``DarC``, under the ``version`` key.

Reviewer: ```suggestion
When compiling Daml project X, you must ensure that the ``DarC`` dependency referenced by both ``DarA`` and ``DarB`` either has identical Daml contents or has a different version number if the contents differ. The version number is defined in the daml.yaml file of the Daml project producing ``DarC``, under the ``version`` key.
```
